### ISSUE TITLE
Feature: Notification center 

### ISSUE THREAD

**Issue by maparr** [spacemeshos/smapp#1068]
### Rationale

I propose to create a tab in smapp , where we will be collecting errors / info messages and announcements of release and new networks for users. Currently, a user has to be subscribed to the Discord announcement channel to get the latest important information for him about smapp or spacemesh. And issues with the application can be shown and hidden in seconds, which can break the app. It will help us to debug the issue, to send the collection of issues with logs to our Sentry.

### Prerequisites

Currently, we have a Discord announcement channel for network and smapp release announcements. For issues in the smapp we have a modal with an issue message which user can close. 

### User story

Scenario with announcements

- [ ] Smapp opened
- [ ] Spacemesh team releases a new network or smapp version 
- [ ] User received a notification about the announcement, can open the notification by deep-link or open notifications in the smapp
- [ ] User can download new sampp by link or switch to the new network in the smapp setting tab

Scenario with critical errors or warnings 

- [ ] Smapp opened
- [ ] During the smeshing process, RPC sends an error, we do push notification about the issue, but the client is on the Network tab and doesn't see the issue from the RPC. He sees the message in the notification tab or in the notification center.
- [ ] Open the notification by clicking, which will redirect you to the correct tab. 
- [ ] User can resolve the problem by restarting the app or asking what to do with this. We can additionally add an link to the wiki page how to resolve such issues with go-spacemesh node ( out of scope, but nice to have )

### Out of scope

wiki documentation or short messages **how to resolve** noties  

### How to implement

For announcements, we can use firebase notifications for the application instance and attach them to the Discord announcement channel. In this case, when a user has an internet, he will receive a new message from Firebase [FCM](https://firebase.google.com/docs/cloud-messaging), [electron approach](https://stackoverflow.com/questions/68329214/can-fcm-push-notifications-in-electron-app),

For critical Error and Warning messages, we can log important messages for our client and push them internally as notifications, which helps to work with and get the latest updates from the app in the notification center,  

<img width="341" alt="Screenshot 2022-12-27 at 22 46 18" src="https://user-images.githubusercontent.com/54288612/209754043-48dc18aa-8b51-47f9-98ed-b7be2fdebf49.png">
<img width="274" alt="Screenshot 2022-12-27 at 22 47 33" src="https://user-images.githubusercontent.com/54288612/209754157-b1a64937-5179-4088-9012-4f10d47afbea.png">

For deep linking, we can use [the official documentation ](https://www.electronjs.org/docs/latest/tutorial/launch-app-from-url-in-another-app), deep link will help to redirect to the correct page even the app is not opened yet

### After the research. We have some goals here 
- local persist module
  - user wants to track when some set up already done, get notifications, and keep the track of main points, which need to be done before smashing 
  
  we can use here [storage](https://github.com/sindresorhus/electron-store), restore for the session's last track of notifications, news, or points for our user
  
- we should be able to rich our users in the app and ask about the app update, and local news ever.
we can use here notification center, I'm researching free solutions for this. [FCM](https://firebase.google.com/pricing)
 no cost for us with [electron push receiver](https://github.com/MatthieuLemoine/electron-push-receiver).
  
Hey folks, this improvement is opend for discussion, we can replace smth or add details. 
cc: @brusherru @monikasmolarek @pigmej

**Comment by pigmej** [comment]
I'd put "new network" aside of that. During mainnet makes ~0 sense IMHO.

New smapp releases seems like standard "new release" problem in software development so yes we should solve that. Probably we should hook up gh releases there too to not have the same information in multiple places.

About the issues, we need to integrate it with network wide status to also avoid dulication of information sources.

**Comment by maparr** [comment]
> I'd put "new network" aside of that. During mainnet makes ~0 sense IMHO.
> 
> New smapp releases seems like standard "new release" problem in software development so yes we should solve that. Probably we should hook up gh releases there too to not have the same information in multiple places.
> 
> About the issues, we need to integrate it with network wide status to also avoid dulication of information sources.

yes, agree to put it aside, and only keep information about the app and what was changed what last news etc. The priority is the state, for example, the network was disconnected at 2PM , but at 3PM connected. 

I can start from simple implementation , POC, and extend if will be ok

**Comment by monikasmolarek** [comment]
@maparr Just as a reminder of what was already discussed but came back in one of the Discord feedbacks - please remember about the desktop notification that is supposed to be shown once the POS creation is complete (the one that is shown on the fireworks confirmation screen when we finish the POS setup steps). 
Or maybe, as a temporary fix, we should hide this info about the expected desktop notification?
